### ISSUE TITLE
performence in wild world

### ISSUE THREAD

**Issue by Vincent630** [Junda24/MonSter#8]
i test kitt model and mixall model in my custom dataset.and it looks not good enough.and the same problem have seen when i use IGEV,compare raft-stereo they both not good enough. maybe i did something wrong when i inference stereo pair

**Comment by Vincent630** [comment]
this is one badcase, left and right ground not good 
![Image](https://github.com/user-attachments/assets/20db34f8-5030-4b4b-9685-d38cc95f5fda)

**Comment by Junda24** [comment]
> this is one badcase, left and right ground not good ![Image](https://github.com/user-attachments/assets/20db34f8-5030-4b4b-9685-d38cc95f5fda)

can you show me the stereo pairs？

**Comment by Vincent630** [comment]
[data.zip](https://github.com/user-attachments/files/19122784/data.zip)

**Comment by Junda24** [comment]
> [data.zip](https://github.com/user-attachments/files/19122784/data.zip)

![Image](https://github.com/user-attachments/assets/76458a20-8237-48fa-9691-b11ab5691a80)
There obviously exists image distortion; these are not strictly calibrated binocular image pairs. All of our training data are trained on strictly calibrated binocular data. Moreover, both IGEV and Monster utilize attention volume to enhance correlation, making them more sensitive to non-calibrated images where the epipolar lines are not on the horizontal line. To achieve good performance in such scenarios, it might be necessary to fine-tune on fisheye images or datasets similar to yours.

**Comment by Vincent630** [comment]
> > [data.zip](https://github.com/user-attachments/files/19122784/data.zip)
> 
> ![Image](https://github.com/user-attachments/assets/76458a20-8237-48fa-9691-b11ab5691a80) There obviously exists image distortion; these are not strictly calibrated binocular image pairs. All of our training data are trained on strictly calibrated binocular data. Moreover, both IGEV and Monster utilize attention volume to enhance correlation, making them more sensitive to non-calibrated images where the epipolar lines are not on the horizontal line. To achieve good performance in such scenarios, it might be necessary to fine-tune on fisheye images or datasets similar to yours. Of course, the most direct way is to dedistort the image.

ok,i see , so that mean after distortion correction of the image pair, i can get a more reasonable output?

**Comment by Junda24** [comment]
> > > [data.zip](https://github.com/user-attachments/files/19122784/data.zip)
> > 
> > 
> > ![Image](https://github.com/user-attachments/assets/76458a20-8237-48fa-9691-b11ab5691a80) There obviously exists image distortion; these are not strictly calibrated binocular image pairs. All of our training data are trained on strictly calibrated binocular data. Moreover, both IGEV and Monster utilize attention volume to enhance correlation, making them more sensitive to non-calibrated images where the epipolar lines are not on the horizontal line. To achieve good performance in such scenarios, it might be necessary to fine-tune on fisheye images or datasets similar to yours. Of course, the most direct way is to dedistort the image.
> 
> ok,i see , so that mean after distortion correction of the image pair, i can get a more reasonable output?

Theoretically, it will definitely be better because the search direction of stereo matching is horizontal. If there is distortion, it will not be able to search for the correct pixels.

**Comment by Vincent630** [comment]
thanks a lot ,that is really helpful,and i will do as you say,and training with my own dataset.

**Comment by Junda24** [comment]
> thanks a lot ,that is really helpful,and i will do as you say,and training with my own dataset.

You're welcome, and thank you for your attention. In fact, you could first try the effect of directly correcting the distortion. I think if the performanceis good enough, there will be no need to retrain.

**Comment by Junda24** [comment]
> > > thanks a lot ,that is really helpful,and i will do as you say,and training with my own dataset.
> > 
> > 
> > You're welcome, and thank you for your attention. In fact, you could first try the effect of directly correcting the distortion. I think if the performanceis good enough, there will be no need to retrain. From the results of the portrait images, the areas with relatively small distortions have fairly good outcomes. It seems that re - finetuning might not be necessary.
> 
> i am back,after calibration my stereo dataset,and perform distortion correction on a set of stereo images using MATLAB.and it seems still not perform well enough.because of no efficent metric depth,we have problems to train monster with our dataset. if it is not inconvenient,i can send you some data which both include distorted and raw images if i can get your email, i will not disturb you without your permission

my WeChat: kkkk24ko
